### PR TITLE
Pfappserver Updates

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue [hunk 1]
```diff
@@ -828,39 +828,53 @@ export default {
...
+          response.items.forEach((item, _index, items) => {
+            let index = this.tableValues.findIndex(node => node.mac === item.mac)
+            this.setRowVariant(index, convert.statusToVariant({ status: item.status }))
+            this.setRowMessage(index, item.message)
+            if (item.message) {
+              this.$store.dispatch('notification/status_' + item.status, {message: this.$i18n.t('Node') + ' ' + item.mac + ': ' + item.message})
+            }
+          })
+          this.$store.dispatch('notification/info', {
+            message: response.items.length + ' ' + this.$i18n.t('node profiling refreshed'),
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue]: incorrect string

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue [hunk 2]
```diff
@@ -828,39 +828,53 @@ export default {
...
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue]: incorrect string

**Comment by satkunas** [comment]
Sample CSV file to test CSV parsing

[nodes.zip](https://github.com/inverse-inc/packetfence/files/2380134/nodes.zip)

File: html/pfappserver/root/static.alt/src/globals/pfFormatters.js [hunk 3]
```diff
@@ -6,15 +6,40 @@ export const pfFormatters = {
   },
   categoryId: (value, key, item) => {
     if (!value) return null
-    return store.state.config.roles.filter(role => role.category_id === item.category_id).map(role => role.name)
+    return store.state.config.roles.filter(role => role.category_id === item.category_id).map(role => role.name)[0]
+  },
+  categoryIdFromIntOrString: (value, key, item) => {
+    if (!value) return null
+    if (isNaN(parseInt(value))) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/pfFormatters.js]: Technically a category name could begin with a number.

File: html/pfappserver/root/static.alt/src/globals/pfFormatters.js [hunk 4]
```diff
@@ -6,15 +6,40 @@ export const pfFormatters = {
...
+      return store.state.config.roles.filter(role => role.name.toLowerCase() === value.toLowerCase()).map(role => role.category_id)[0] // string
+    } else {
+      return value // int
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/pfFormatters.js]: Not true. Ex: `parseInt('1st') === 1`
Depending on the strategy you use, you may need to return `parseInt(value)`.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 5]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+    let body = {
+      cursor: state.searchPageSize * (page - 1),
+      limit: state.searchPageSize,
+      fields: state.searchFields,
+      sort
+    }
+    let apiPromise = state.searchQuery ? api.search(Object.assign(body, {query: state.searchQuery})) : api.all(body)
+    if (state.searchStatus !== 'loading') {
+      return new Promise((resolve, reject) => {
+        commit('SEARCH_REQUEST')
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 6]
```diff
@@ -9,13 +9,15 @@ const STORAGE_SAVED_SEARCH = 'users-saved-search'
...
+  userExists: {}, // node exists true|false
   message: '',
   userStatus: '',
   savedSearches: JSON.parse(localStorage.getItem(STORAGE_SAVED_SEARCH)) || []
 }
 
 const getters = {
-  isLoading: state => state.userStatus === 'loading'
+  isLoading: state => state.userStatus === 'loading',
+  isLoadingResults: state => state.searchStatus === 'loading'
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This state doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 7]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+      limit: state.searchPageSize,
+      fields: state.searchFields,
+      sort
+    }
+    let apiPromise = state.searchQuery ? api.search(Object.assign(body, {query: state.searchQuery})) : api.all(body)
+    if (state.searchStatus !== 'loading') {
+      return new Promise((resolve, reject) => {
+        commit('SEARCH_REQUEST')
+        apiPromise.then(response => {
+          commit('SEARCH_SUCCESS', response)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 8]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+          resolve(response)
+        }).catch(err => {
+          commit('SEARCH_ERROR', err.response)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 9]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+          reject(err)
+        })
+      })
+    }
+  },
+  exists: ({commit}, pid) => {
+    if (state.userExists[pid] !== undefined) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: I suggest this way:
```js
if (state.userExists.hasOwnProperty(pid)) {
```

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 10]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+      limit: 1,
+      query: {
+        op: 'and',
+        values: [{
+          field: 'pid', op: 'equals', value: pid
+        }]
+      }
+    }
+    return new Promise((resolve, reject) => {
+      commit('SEARCH_REQUEST')
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 11]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+      api.search(body).then(response => {
+        commit('SEARCH_SUCCESS')
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 12]
```diff
@@ -34,6 +36,62 @@ const actions = {
...
+        if (response.items.length > 0) {
+          commit('USER_EXISTS', pid)
+          resolve(true)
+        } else {
+          commit('USER_NOT_EXISTS', pid)
+          reject(new Error('Unknown PID'))
+        }
+      }).catch(err => {
+        commit('SEARCH_ERROR', err.response)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: This mutation doesn't exist.

File: html/pfappserver/root/static.alt/src/components/pfCSVParse.vue [hunk 13]
```diff
@@ -0,0 +1,923 @@
...
+        minLength: minLength(1),
+        $each: eachExportModel
+      }
+    }
+  },
+  methods: {
+    uuidStr (section) {
+      return (section || 'default') + '-' + this.uuid
+    },
+    parse () {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfCSVParse.vue]: Use a [debouncer](https://www.npmjs.com/package/promised-debounce).

File: html/pfappserver/root/static.alt/src/components/pfCSVParse.vue [hunk 14]
```diff
@@ -0,0 +1,923 @@
...
+      pageSizeLimit: 100
+    }
+  },
+  validations () {
+    // dynamically apply vuelidate validations
+    let eachSelectValues = {}
+    let eachStaticMapping = {}
+    let eachExportModel = {}
+
+    this.fields.forEach((field, index, fields) => {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfCSVParse.vue]: The `fields` argument doesn't seem necessary.

**Review by cgx — DISMISSED** [review]

File: html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js [hunk 15]
```diff
@@ -36,30 +35,8 @@ const actions = {
...
-        }).catch(err => {
-          commit('SEARCH_ERROR', err.response)
-          reject(err)
-        })
-      })
-    }
-  },
   exists: ({commit}, mac) => {
-    if (state.nodeExists[mac] !== undefined) {
+    if (state.nodeExists.hasOwnProperty('mac')) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js]: Use the `mac` variable (not a static string).

File: html/pfappserver/root/static.alt/src/views/Users/_store/index.js [hunk 16]
```diff
@@ -36,30 +35,8 @@ const actions = {
...
-        }).catch(err => {
-          commit('SEARCH_ERROR', err.response)
-          reject(err)
-        })
-      })
-    }
-  },
   exists: ({commit}, pid) => {
-    if (state.userExists[pid] !== undefined) {
+    if (state.userExists.hasOwnProperty('pid')) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/_store/index.js]: Use the `pid` variable (not a static string).

**Pull request merged**